<그룹>
- R*: 테스트베드를 통과하여 상용서비스가 가능한 알고리즘
- R+: 테스트베드 심사 중이거나 운용심사만 완료한 알고리즘
- R!: 관심 또는 주의가 필요한 알고리즘

<용어>
- 표준편차: 수익률의 변동성을 나타내는 지표로 값이 클수록 변동성이 심하므로 위험이 크고, 값이 작을수록 위험이 작다고 할 수 있습니다.

- 샤프지수: 위험 1단위에 대한 초과수익의 정도를 나타내는 지표로써, 샤프지수가 높을수록 투자성과가 성공적이라고 할 수 있습니다. 단, 측정 기간과 투자 전략이 동일한 경우의 비교시에 의미가 있습니다.

- 최대손실률: 운용 기간 중 최고가를 기준으로 발생한 최대 손실의 비율을 표시합니다. 향후 투자자가 감내해야 할 위험의 수준을 알려주는 지표로 최대손실률이 적을 수록 안정적입니다.

- 보상비율: 최종손익을 최대손실폭으로 나눈 값으로 1미만인 경우 위험도에 비해 수익이 적고, 1이상인 경우 위험도에 비해 수익이 높고 안정적이라 볼 수 있습니다.

- 자산위험도: 자산군별 위험정도에 대한 값으로 0 ~ 10범위로 표시합니다. 값이 클수록 위험도가 높거나 투자에 적극적인 자산으로 구성되어 있음을 나타냅니다.
            자산군별 위험도의 값을 '주식 > 주식ETF > ELS/DLS > ETN > 펀드 > 기타상품 > 기타ETF > 채권ETF > RP' 순으로 설정하여 계산하였습니다.

- 유형차별성: 안정추구형, 위험중립형, 적극투자형 세가지 유형의 차별성을 나타내는 지표로 0 ~ 10 범위이며 값이 클수록 차별성이 높다고 할 수 있습니다.

- 종목편중률: 포트폴리오 구성종목 중 가장 높은 비중의 종목에 대한 자산군(색상)과 비율(%)을 표시합니다.(※ 예금,RP 등 현금성 자산은 제외합니다.)
            (※ 자산군 색상표시 :주식주식ETFELS/DLS ETN펀드기타상품 기타ETF채권ETF)

- 다계좌운용: 동일 유형 내 3개 계좌간 방향성의 차이를 나타낸 것으로, 값이 작을수록 다계좌 운용능력이 좋다고 할 수 있습니다. 다계좌운용지표는 현재 심사 중인 알고리즘에 대해서만 산출합니다.

- 자산별 투자비중: 자산군 분류기준(주식/펀드/주식ETF/채권ETF/기타ETF/ETN/예금/기타)에 따른 구분으로 자산 배분상태 확인이 가능합니다. RP는 예금으로, 해외ETF는 기타ETF로 분류됩니다.
            (※ 해외자산을 포함하는 포트폴리오의 운용실적은 환율변동에 의해 영향을 받을 수 있습니다.)

In [77]:
import requests
from bs4 import BeautifulSoup
import pandas as pd
pd.set_option('display.max_columns', 27)
import re

url = "https://www.ratestbed.kr:7443/portal/pblntf/listProgrsInfo2.do?menuNo=200238"

res = requests.get(url)
soup = BeautifulSoup(res.text, "lxml")

date = soup.select_one('span.fc_point').text # 운용기준일
print(date)

2023년 07월 06일


In [78]:
columns_list = ['그룹', '알고리즘명', '유형', '업체명', '운용자금(원)',
                '1주', '1개월', '3개월', '6개월', '1년', '2년', '3년', '연환산', '누적',
                '표준편차', '샤프지수', '최대손실률', '보상비율',
                '자산 위험도', '유형 차별성', '종목 편중률', '편중된 종목', '다계좌 운용', '구성 종목수', '매매 회전률',
                '공시시작일', '공시기간']

ra_df = pd.DataFrame(columns=columns_list)

ra_df

Empty DataFrame
Columns: [그룹, 알고리즘명, 유형, 업체명, 운용자금(원), 1주, 1개월, 3개월, 6개월, 1년, 2년, 3년, 연환산, 누적, 표준편차, 샤프지수, 최대손실률, 보상비율, 자산 위험도, 유형 차별성, 종목 편중률, 편중된 종목, 다계좌 운용, 구성 종목수, 매매 회전률, 공시시작일, 공시기간]
Index: []

In [79]:
# 기본 정보
ra_df['그룹'] = [n.text for n in soup.select('p.name span.league_sym')] # 그룹
ra_df['알고리즘명'] = [n.text for n in soup.select('p.name span:nth-child(2)')] # 알고리즘명
ra_df['유형'] = [n.text for n in soup.select('p.clear span')] # 유형
ra_df['유형'] = ra_df['유형'].apply(lambda x: x.replace('[', '').replace(']', ''))
ra_df['업체명'] = [n.text for n in soup.select('td.logo span')] # 업체명
ra_df['운용자금(원)'] = [n.text.replace('\r', '').replace('\t', '').replace('\n', '').replace(',', '') for n in soup.select('td.amount')] # 운용자금(원)

# 수익률(%)
ra_df['1주'] = [n.text for n in soup.select('td.tar.profit1w')] # 수익률 1주
ra_df['1개월'] = [n.text for n in soup.select('td.tar.profit1m')] # 수익률 1달
ra_df['3개월'] = [n.text for n in soup.select('td.tar.profit3m')] # 수익률 3달
ra_df['6개월'] = [n.text for n in soup.select('td.tar.profit6m')] # 수익률 6달
ra_df['1년'] = [n.text for n in soup.select('td.tar.profit1y')] # 수익률 1년
ra_df['2년'] = [n.text for n in soup.select('td.tar.profit2y')] # 수익률 2년
ra_df['3년'] = [n.text for n in soup.select('td.tar.profit3y')] # 수익률 3년
ra_df['연환산'] = [n.text for n in soup.select('td.tar.profityc')] # 수익률 연환산
ra_df['누적'] = [n.text for n in soup.select('td.tar.profit')] # 수익률 누적

# 위험지표
ra_df['표준편차'] = [n.text for n in soup.select('td.tar.stdev')] # 표준편자
ra_df['샤프지수'] = [n.text for n in soup.select('td.tar.sharpe')] # 샤프지수
ra_df['최대손실률'] = [n.text for n in soup.select('td.tar.maxdd')] # 최대손실률
ra_df['보상비율'] = [n.text for n in soup.select('td.tar.reward')] # 보상비율

# 참고지표
ra_df['자산 위험도'] = [n.text for n in soup.select('td.tar.active')] # 자산위험도
ra_df['유형 차별성'] = [n.text for n in soup.select('td.tar.diff')] # 유형차별성
ra_df['종목 편중률'] = [n.text for n in soup.select('span.maxasset')] # 종목편중률
ra_df['편중된 종목'] = [n.attrs['title'] for n in soup.find_all('span', class_='maxasset')] # 편중된 종목
ra_df['다계좌 운용'] = [n.text.replace('\r', '').replace('\t', '').replace('\n', '') for n in soup.select('td.multiacc')] # 다계좌운용
ra_df['구성 종목수'] = [n.text for n in soup.select('td.tar.issuecnt')] # 구성종목수
ra_df['매매 회전률'] = [n.text for n in soup.select('td.tar.turnover')] # 매매회전율
ra_df['공시시작일'] = [n.text for n in soup.select('td[rowspan="1"]')] # 공시시작일
ra_df['공시기간'] = [n.text for n in soup.select('span[class="optermText"]')] # 공시시간

ra_df.head()

그룹                      알고리즘명                  유형      업체명   운용자금(원)     1주  \
0  *  가우스앤 밸류핀(GaussN ValueFin)  [메타로고스밸류핀(적극투자형1)]    메타로고스  30000000  -0.56   
1  *  가우스앤 밸류핀(GaussN ValueFin)  [메타로고스밸류핀(안정추구형3)]    메타로고스  60000000  -0.29   
2  *  가우스앤 밸류핀(GaussN ValueFin)  [메타로고스밸류핀(위험중립형2)]    메타로고스  45000000  -1.16   
3  *              디셈버 미국 주식 솔루션      [주식회사 디셈버앤컴퍼니]  디셈버앤컴퍼니  45000000  -0.47   
4  *              디셈버 미국 주식 솔루션      [주식회사 디셈버앤컴퍼니]  디셈버앤컴퍼니  45000000   0.28   

     1개월    3개월    6개월     1년     2년     3년    연환산     누적  표준편차   샤프지수  \
0   0.17   2.46   2.82   1.20  -4.83          2.43   5.03  0.07   0.05   
1  -0.26   0.98   2.47   0.93  -3.66         -0.09  -0.67  0.04  -0.00   
2  -1.87  -0.44   1.30  -0.75  -4.36          0.82   1.29  0.05  -0.00   
3  -0.13   1.08   6.70   5.55  18.41  20.76   5.40  37.10  0.07   0.54   
4   0.65   3.20  11.56  10.40  22.43  43.15   7.78  59.22  0.11   0.55   

    최대손실률   보상비율 자산 위험도 유형 차별성 종목 편중률   편중된 종목 다계좌 운용 구성 종목수   매매 회전률  \
0  -12.95   0.35   6.23   7.04  23.54  국내기타ETF      -      7   530.14   
1   -8.19  -0.08   4.47   7.04  31.23  국내채권ETF      -      7   330.80   
2   -8.79   0.14   5.19   7.04  21.28  국내기타ETF      -      7   467.66   
3  -11.81   2.20   5.92   8.10  43.92  해외기타ETF      -     26  1287.36   
4  -21.50   2.32   7.84   8.10  21.79  해외기타ETF      -     26  2388.00   

        공시시작일  공시기간  
0  2021-02-01  2.4년  
1  2021-02-01  2.4년  
2  2021-02-01  2.4년  
3  2017-05-22  6.1년  
4  2017-05-22  6.1년

In [80]:
ra_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 689 entries, 0 to 688
Data columns (total 27 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   그룹       689 non-null    object
 1   알고리즘명    689 non-null    object
 2   유형       689 non-null    object
 3   업체명      689 non-null    object
 4   운용자금(원)  689 non-null    object
 5   1주       689 non-null    object
 6   1개월      689 non-null    object
 7   3개월      689 non-null    object
 8   6개월      689 non-null    object
 9   1년       689 non-null    object
 10  2년       689 non-null    object
 11  3년       689 non-null    object
 12  연환산      689 non-null    object
 13  누적       689 non-null    object
 14  표준편차     689 non-null    object
 15  샤프지수     689 non-null    object
 16  최대손실률    689 non-null    object
 17  보상비율     689 non-null    object
 18  자산 위험도   689 non-null    object
 19  유형 차별성   689 non-null    object
 20  종목 편중률   689 non-null    object
 21  편중된 종목   689 non-null    object
 22  다계

### 정렬하기 위해서 시리즈 타입을 숫자로 바꾸기

In [81]:
for column in ra_df.columns[4:21]:
    ra_df[column] = pd.to_numeric(ra_df[column], errors='coerce')

for column in ra_df.columns[22:25]:
    ra_df[column] = pd.to_numeric(ra_df[column], errors='coerce')

ra_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 689 entries, 0 to 688
Data columns (total 27 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   그룹       689 non-null    object 
 1   알고리즘명    689 non-null    object 
 2   유형       689 non-null    object 
 3   업체명      689 non-null    object 
 4   운용자금(원)  686 non-null    float64
 5   1주       686 non-null    float64
 6   1개월      686 non-null    float64
 7   3개월      518 non-null    float64
 8   6개월      518 non-null    float64
 9   1년       416 non-null    float64
 10  2년       240 non-null    float64
 11  3년       195 non-null    float64
 12  연환산      518 non-null    float64
 13  누적       686 non-null    float64
 14  표준편차     686 non-null    float64
 15  샤프지수     686 non-null    float64
 16  최대손실률    686 non-null    float64
 17  보상비율     686 non-null    float64
 18  자산 위험도   686 non-null    float64
 19  유형 차별성   686 non-null    float64
 20  종목 편중률   680 non-null    float64
 21  편중된 종목   689 non

In [82]:
ra_df.isnull().sum()

그룹           0
알고리즘명        0
유형           0
업체명          0
운용자금(원)      3
1주           3
1개월          3
3개월        171
6개월        171
1년         273
2년         449
3년         494
연환산        171
누적           3
표준편차         3
샤프지수         3
최대손실률        3
보상비율         3
자산 위험도       3
유형 차별성       3
종목 편중률       9
편중된 종목       0
다계좌 운용     500
구성 종목수       9
매매 회전률       6
공시시작일        0
공시기간         0
dtype: int64

In [83]:
ra_df.head()

그룹                      알고리즘명                  유형      업체명     운용자금(원)  \
0  *  가우스앤 밸류핀(GaussN ValueFin)  [메타로고스밸류핀(적극투자형1)]    메타로고스  30000000.0   
1  *  가우스앤 밸류핀(GaussN ValueFin)  [메타로고스밸류핀(안정추구형3)]    메타로고스  60000000.0   
2  *  가우스앤 밸류핀(GaussN ValueFin)  [메타로고스밸류핀(위험중립형2)]    메타로고스  45000000.0   
3  *              디셈버 미국 주식 솔루션      [주식회사 디셈버앤컴퍼니]  디셈버앤컴퍼니  45000000.0   
4  *              디셈버 미국 주식 솔루션      [주식회사 디셈버앤컴퍼니]  디셈버앤컴퍼니  45000000.0   

     1주   1개월   3개월    6개월     1년     2년     3년   연환산     누적  표준편차  샤프지수  \
0 -0.56  0.17  2.46   2.82   1.20  -4.83    NaN  2.43   5.03  0.07  0.05   
1 -0.29 -0.26  0.98   2.47   0.93  -3.66    NaN -0.09  -0.67  0.04 -0.00   
2 -1.16 -1.87 -0.44   1.30  -0.75  -4.36    NaN  0.82   1.29  0.05 -0.00   
3 -0.47 -0.13  1.08   6.70   5.55  18.41  20.76  5.40  37.10  0.07  0.54   
4  0.28  0.65  3.20  11.56  10.40  22.43  43.15  7.78  59.22  0.11  0.55   

   최대손실률  보상비율  자산 위험도  유형 차별성  종목 편중률   편중된 종목  다계좌 운용  구성 종목수   매매 회전률  \
0 -12.95  0.35    6.23    7.04   23.54  국내기타ETF     NaN     7.0   530.14   
1  -8.19 -0.08    4.47    7.04   31.23  국내채권ETF     NaN     7.0   330.80   
2  -8.79  0.14    5.19    7.04   21.28  국내기타ETF     NaN     7.0   467.66   
3 -11.81  2.20    5.92    8.10   43.92  해외기타ETF     NaN    26.0  1287.36   
4 -21.50  2.32    7.84    8.10   21.79  해외기타ETF     NaN    26.0  2388.00   

        공시시작일  공시기간  
0  2021-02-01  2.4년  
1  2021-02-01  2.4년  
2  2021-02-01  2.4년  
3  2017-05-22  6.1년  
4  2017-05-22  6.1년

In [85]:
ra_df.describe()

운용자금(원)          1주         1개월         3개월         6개월  \
count  6.860000e+02  686.000000  686.000000  518.000000  518.000000   
mean   8.410967e+06    0.630131    1.161137    2.107761    7.883919   
std    1.128882e+07    4.169827    5.185901    4.463134    8.755677   
min    1.000000e+05   -4.400000  -32.700000  -13.510000  -10.380000   
25%    2.000000e+06   -0.175000   -0.070000   -0.357500    3.357500   
50%    5.000000e+06    0.260000    0.650000    1.555000    6.005000   
75%    1.000000e+07    0.687500    1.640000    3.575000    9.715000   
max    6.500000e+07   69.840000   74.620000   37.610000   60.550000   

               1년          2년          3년         연환산          누적        표준편차  \
count  416.000000  240.000000  195.000000  518.000000  686.000000  686.000000   
mean     4.113582   -0.994917   24.487282    2.403166    9.465058    0.152128   
std      7.101509   11.680790   30.020189   21.365120   25.906983    0.307966   
min    -17.460000  -36.050000   -3.500000 -249.540000  -33.840000    0.000000   
25%      0.457500   -7.030000    8.970000   -0.785000   -1.312500    0.060000   
50%      3.320000   -2.170000   14.610000    2.415000    2.215000    0.090000   
75%      7.040000    4.787500   27.025000    5.372500   14.692500    0.130000   
max     43.010000   36.380000  248.140000  109.360000  415.230000    3.290000   

             샤프지수       최대손실률        보상비율      자산 위험도      유형 차별성      종목 편중률  \
count  686.000000  686.000000  686.000000  686.000000  686.000000  680.000000   
mean     0.294373  -16.245700    0.643207    5.184767    6.799082   25.469176   
std      0.815804   18.071369    2.079053    1.955087    1.494948   17.022167   
min     -5.720000  -98.270000   -0.960000    0.000000    2.500000    1.540000   
25%      0.000000  -19.487500   -0.130000    3.982500    5.800000   10.887500   
50%     -0.000000  -11.800000    0.260000    5.010000    7.030000   22.285000   
75%      0.320000   -5.307500    0.887500    6.087500    7.890000   34.255000   
max      5.010000    0.000000   29.410000    9.860000    9.870000   82.420000   

           다계좌 운용      구성 종목수       매매 회전률  
count  189.000000  680.000000   683.000000  
mean     0.121270   11.807353   523.924451  
std      0.213249    8.156402   599.270975  
min      0.000000    3.000000     3.450000  
25%      0.020000    7.000000   138.090000  
50%      0.060000   10.000000   320.890000  
75%      0.140000   13.000000   705.365000  
max      1.660000  101.000000  7663.690000

In [86]:
ra_df['그룹'].unique()

array(['*', '+', '!'], dtype=object)

### 파운트투자자문 R* 그룹 알고리즘 확인_6개

In [94]:
ra_df.loc[(ra_df['그룹'] == '*') & (ra_df['업체명'] == '파운트투자자문')].sort_values(by='누적', ascending=False)

그룹             알고리즘명     유형      업체명    운용자금(원)    1주   1개월   3개월   6개월  \
337  *   파운트 블루웨일(펀드)1.0  적극투자형  파운트투자자문   300000.0  1.62  4.45  3.24  6.23   
339  *   파운트 블루웨일(펀드)1.0  위험중립형  파운트투자자문   300000.0  1.38  3.81  2.73  5.27   
338  *   파운트 블루웨일(펀드)1.0  안정추구형  파운트투자자문   300000.0  0.89  2.80  2.44  4.36   
95   *  파운트 블루웨일(ETF)1.0  안정추구형  파운트투자자문  5000000.0 -0.89 -0.50 -2.04  1.97   
96   *  파운트 블루웨일(ETF)1.0  위험중립형  파운트투자자문  5000000.0 -0.76 -0.37 -1.89  1.73   
94   *  파운트 블루웨일(ETF)1.0  적극투자형  파운트투자자문  5000000.0 -0.74 -0.32 -2.16  0.26   

       1년     2년    3년   연환산     누적  표준편차  샤프지수  최대손실률  보상비율  자산 위험도  유형 차별성  \
337  0.86 -13.97  6.42  3.54  20.95  0.13  0.14 -22.30  0.64    5.70    7.11   
339  1.02 -11.38  4.29  3.01  17.57  0.15  0.09 -27.95  0.41    5.70    7.11   
338  2.57  -6.16  6.52  2.98  17.16  0.07  0.19 -11.70  1.16    5.70    7.11   
95  -1.46  -1.76 -2.03 -0.06  -1.53  0.08 -0.00 -12.90 -0.11    4.85    6.47   
96  -1.80  -3.75 -1.74 -0.24  -2.30  0.10 -0.00 -17.87 -0.12    4.73    6.47   
94  -2.94  -4.83 -2.24 -0.69  -4.45  0.14 -0.00 -28.06 -0.12    4.70    6.47   

     종목 편중률   편중된 종목  다계좌 운용  구성 종목수   매매 회전률       공시시작일  공시기간  
337   35.20     국내펀드     NaN     5.0  1395.85  2018-05-08  5.2년  
339   30.20     국내펀드     NaN     5.0  1297.44  2018-05-08  5.2년  
338   29.73     국내펀드     NaN     5.0  1177.13  2018-05-08  5.2년  
95    41.69  해외기타ETF     NaN     7.0  1457.66  2018-05-08  5.2년  
96    31.70  해외기타ETF     NaN     8.0  1534.55  2018-05-08  5.2년  
94    27.08  해외기타ETF     NaN     8.0  1601.13  2018-05-08  5.2년

In [99]:
ra_df.loc[(ra_df['그룹'] == '*') & (ra_df['업체명'] == '파운트투자자문')].describe()

운용자금(원)        1주       1개월       3개월       6개월        1년  \
count  6.000000e+00  6.000000  6.000000  6.000000  6.000000  6.000000   
mean   2.650000e+06  0.250000  1.645000  0.386667  3.303333 -0.291667   
std    2.574296e+06  1.171597  2.298336  2.661065  2.326574  2.092352   
min    3.000000e+05 -0.890000 -0.500000 -2.160000  0.260000 -2.940000   
25%    3.000000e+05 -0.755000 -0.357500 -2.002500  1.790000 -1.715000   
50%    2.650000e+06  0.075000  1.240000  0.275000  3.165000 -0.300000   
75%    5.000000e+06  1.257500  3.557500  2.657500  5.042500  0.980000   
max    5.000000e+06  1.620000  4.450000  3.240000  6.230000  2.570000   

              2년        3년       연환산        누적      표준편차      샤프지수      최대손실률  \
count   6.000000  6.000000  6.000000   6.00000  6.000000  6.000000   6.000000   
mean   -6.975000  1.870000  1.423333   7.90000  0.111667  0.070000 -20.130000   
std     4.715849  4.320083  1.941862  11.79022  0.033116  0.082946   7.172051   
min   -13.970000 -2.240000 -0.690000  -4.45000  0.070000 -0.000000 -28.060000   
25%   -10.075000 -1.957500 -0.195000  -2.10750  0.085000  0.000000 -26.537500   
50%    -5.495000  1.275000  1.460000   7.81500  0.115000  0.045000 -20.085000   
75%    -4.020000  5.887500  3.002500  17.46750  0.137500  0.127500 -14.142500   
max    -1.760000  6.520000  3.540000  20.95000  0.150000  0.190000 -11.700000   

           보상비율    자산 위험도    유형 차별성     종목 편중률  다계좌 운용    구성 종목수       매매 회전률  
count  6.000000  6.000000  6.000000   6.000000     0.0  6.000000     6.000000  
mean   0.310000  5.230000  6.790000  32.600000     NaN  6.333333  1410.626667  
std    0.526802  0.517301  0.350542   5.189709     NaN  1.505545   155.827412  
min   -0.120000  4.700000  6.470000  27.080000     NaN  5.000000  1177.130000  
25%   -0.117500  4.760000  6.470000  29.847500     NaN  5.000000  1322.042500  
50%    0.150000  5.275000  6.790000  30.950000     NaN  6.000000  1426.755000  
75%    0.582500  5.700000  7.110000  34.325000     NaN  7.750000  1515.327500  
max    1.160000  5.700000  7.110000  41.690000     NaN  8.000000  1601.130000

### 디셈버앤컴퍼니(핀트) R* 그룹 알고리즘 확인_9개

In [95]:
ra_df.loc[(ra_df['그룹'] == '*') & (ra_df['업체명'] == '디셈버앤컴퍼니')].sort_values(by='누적', ascending=False)

그룹               알고리즘명            유형      업체명     운용자금(원)    1주   1개월  \
5    *       디셈버 미국 주식 솔루션  주식회사 디셈버앤컴퍼니  디셈버앤컴퍼니  45000000.0  1.04  1.44   
4    *       디셈버 미국 주식 솔루션  주식회사 디셈버앤컴퍼니  디셈버앤컴퍼니  45000000.0  0.28  0.65   
276  *  디셈버 ISAAC 자산배분 해외형        적극투자형1  디셈버앤컴퍼니   5000000.0  0.53  1.65   
275  *  디셈버 ISAAC 자산배분 해외형        위험중립형1  디셈버앤컴퍼니   5000000.0 -0.16  0.62   
3    *       디셈버 미국 주식 솔루션  주식회사 디셈버앤컴퍼니  디셈버앤컴퍼니  45000000.0 -0.47 -0.13   
90   *  디셈버 ISAAC 자산배분 국내형        적극투자형1  디셈버앤컴퍼니   5000000.0  0.82  1.68   
89   *  디셈버 ISAAC 자산배분 국내형        위험중립형1  디셈버앤컴퍼니   5000000.0  0.45  1.01   
274  *  디셈버 ISAAC 자산배분 해외형        안정추구형1  디셈버앤컴퍼니   5000000.0 -0.88 -0.51   
88   *  디셈버 ISAAC 자산배분 국내형        안정추구형1  디셈버앤컴퍼니   5000000.0  0.17  0.38   

      3개월    6개월     1년     2년     3년   연환산     누적  표준편차  샤프지수  최대손실률  보상비율  \
5    5.31  16.15  14.87  26.09  67.66  9.56  78.12  0.18  0.43 -34.33  1.89   
4    3.20  11.56  10.40  22.43  43.15  7.78  59.22  0.11  0.55 -21.50  2.32   
276  2.63   9.28   6.54   7.63  25.57  6.02  49.24  0.11  0.40 -21.90  1.78   
275  1.00   6.98   3.43   7.59  16.58  4.95  38.20  0.08  0.40 -16.16  1.95   
3    1.08   6.70   5.55  18.41  20.76  5.40  37.10  0.07  0.54 -11.81  2.20   
90   1.84   7.07   4.16  -2.49  14.81  3.60  27.63  0.12  0.17 -25.89  0.87   
89   0.66   4.83   2.84  -1.75   8.97  2.82  21.09  0.07  0.16 -16.17  1.12   
274 -0.84   4.47   0.79   7.06   6.20  2.82  19.13  0.07  0.16 -11.59  1.30   
88   0.00   2.82   2.98  -0.19   3.47  1.54  11.18  0.03 -0.00  -7.04  1.40   

     자산 위험도  유형 차별성  종목 편중률   편중된 종목  다계좌 운용  구성 종목수   매매 회전률       공시시작일  \
5      9.71    8.10   10.82  해외기타ETF     NaN    23.0  3480.91  2017-05-22   
4      7.84    8.10   21.79  해외기타ETF     NaN    26.0  2388.00  2017-05-22   
276    4.99    7.43   20.47  해외기타ETF     NaN    20.0  2235.22  2016-10-17   
275    4.99    7.43   12.90  해외기타ETF     NaN    22.0  1818.85  2016-10-17   
3      5.92    8.10   43.92  해외기타ETF     NaN    26.0  1287.36  2017-05-22   
90     3.98    7.67   19.75  국내기타ETF     NaN    16.0  2162.02  2016-10-17   
89     3.41    7.67   11.97  국내기타ETF     NaN    18.0  1680.49  2016-10-17   
274    4.99    7.43   37.52  해외기타ETF     NaN    16.0  1400.84  2016-10-17   
88     2.49    7.67   28.20  국내채권ETF     NaN    15.0  1321.06  2016-10-17   

     공시기간  
5    6.1년  
4    6.1년  
276  6.7년  
275  6.7년  
3    6.1년  
90   6.7년  
89   6.7년  
274  6.7년  
88   6.7년

In [100]:
ra_df.loc[(ra_df['그룹'] == '*') & (ra_df['업체명'] == '디셈버앤컴퍼니')].describe()

운용자금(원)        1주       1개월       3개월        6개월         1년  \
count  9.000000e+00  9.000000  9.000000  9.000000   9.000000   9.000000   
mean   1.833333e+07  0.197778  0.754444  1.653333   7.762222   5.728889   
std    2.000000e+07  0.613754  0.770083  1.852181   4.079877   4.389050   
min    5.000000e+06 -0.880000 -0.510000 -0.840000   2.820000   0.790000   
25%    5.000000e+06 -0.160000  0.380000  0.660000   4.830000   2.980000   
50%    5.000000e+06  0.280000  0.650000  1.080000   6.980000   4.160000   
75%    4.500000e+07  0.530000  1.440000  2.630000   9.280000   6.540000   
max    4.500000e+07  1.040000  1.680000  5.310000  16.150000  14.870000   

              2년         3년       연환산         누적      표준편차      샤프지수  \
count   9.000000   9.000000  9.000000   9.000000  9.000000  9.000000   
mean    9.420000  23.018889  4.943333  37.878889  0.093333  0.312222   
std    10.601066  20.557698  2.577120  21.377069  0.042720  0.194215   
min    -2.490000   3.470000  1.540000  11.180000  0.030000 -0.000000   
25%    -0.190000   8.970000  2.820000  21.090000  0.070000  0.160000   
50%     7.590000  16.580000  4.950000  37.100000  0.080000  0.400000   
75%    18.410000  25.570000  6.020000  49.240000  0.110000  0.430000   
max    26.090000  67.660000  9.560000  78.120000  0.180000  0.550000   

           최대손실률      보상비율    자산 위험도    유형 차별성     종목 편중률  다계좌 운용     구성 종목수  \
count   9.000000  9.000000  9.000000  9.000000   9.000000     0.0   9.000000   
mean  -18.487778  1.647778  5.368889  7.733333  23.037778     NaN  20.222222   
std     8.370334  0.498768  2.228797  0.293981  11.543151     NaN   4.265495   
min   -34.330000  0.870000  2.490000  7.430000  10.820000     NaN  15.000000   
25%   -21.900000  1.300000  3.980000  7.430000  12.900000     NaN  16.000000   
50%   -16.170000  1.780000  4.990000  7.670000  20.470000     NaN  20.000000   
75%   -11.810000  1.950000  5.920000  8.100000  28.200000     NaN  23.000000   
max    -7.040000  2.320000  9.710000  8.100000  43.920000     NaN  26.000000   

            매매 회전률  
count     9.000000  
mean   1974.972222  
std     696.624882  
min    1287.360000  
25%    1400.840000  
50%    1818.850000  
75%    2235.220000  
max    3480.910000

### DB 적재

In [4]:
import pymysql
import yaml
import sqlalchemy
from sqlalchemy import create_engine, select, text
pymysql.install_as_MySQLdb()

# DB 접속 정보
with open('/mnt/FE0A5E240A5DDA6B/workspace/Quant_Portfolio/Analyst/config.yaml', encoding='UTF-8') as f:
    _cfg = yaml.load(f, Loader=yaml.FullLoader)

DB_SECRET = _cfg['DB_SECRET']

# MySQL에 DF 밀어넣는 함수
def df2sql(is_data: pd.DataFrame, db_table):

    engine = sqlalchemy.create_engine(f"mysql+pymysql://root:{DB_SECRET}@localhost:3306/quant_db?charset=utf8", encoding='utf8')
    conn = engine.connect()

    is_data.to_sql(name=db_table, con=engine, if_exists='replace', index=False) # if_exists='append'
    conn.close()

# df2sql(ra_df, 'tb_RAtestbed')

### sqlalchemy로 불러오기 시도

In [50]:
engine = sqlalchemy.create_engine(f"mysql+pymysql://root:{DB_SECRET}@localhost:3306/quant_db?charset=utf8", encoding='utf8')

with engine.connect() as conn:
    result = conn.execute(text("SELECT * FROM quant_db.tb_RAtestbed WHERE 그룹 = '*' and 업체명 = '파운트투자자문';"))
    for row in result:
        print(row) # <class 'sqlalchemy.engine.row.LegacyRow'>

('*', '파운트 블루웨일(ETF)1.0', '적극투자형', '파운트투자자문', 5000000.0, -0.74, -0.32, -2.16, 0.26, -2.94, -4.83, -2.24, -0.69, -4.45, 0.14, -0.0, -28.06, -0.12, 4.7, 6.47, 27.08, '해외기타ETF', None, 8.0, 1601.13, '2018-05-08', '5.2년')
('*', '파운트 블루웨일(ETF)1.0', '안정추구형', '파운트투자자문', 5000000.0, -0.89, -0.5, -2.04, 1.97, -1.46, -1.76, -2.03, -0.06, -1.53, 0.08, -0.0, -12.9, -0.11, 4.85, 6.47, 41.69, '해외기타ETF', None, 7.0, 1457.66, '2018-05-08', '5.2년')
('*', '파운트 블루웨일(ETF)1.0', '위험중립형', '파운트투자자문', 5000000.0, -0.76, -0.37, -1.89, 1.73, -1.8, -3.75, -1.74, -0.24, -2.3, 0.1, -0.0, -17.87, -0.12, 4.73, 6.47, 31.7, '해외기타ETF', None, 8.0, 1534.55, '2018-05-08', '5.2년')
('*', '파운트 블루웨일(펀드)1.0', '적극투자형', '파운트투자자문', 300000.0, 1.62, 4.45, 3.24, 6.23, 0.86, -13.97, 6.42, 3.54, 20.95, 0.13, 0.14, -22.3, 0.64, 5.7, 7.11, 35.2, '국내펀드', None, 5.0, 1395.85, '2018-05-08', '5.2년')
('*', '파운트 블루웨일(펀드)1.0', '안정추구형', '파운트투자자문', 300000.0, 0.89, 2.8, 2.44, 4.36, 2.57, -6.16, 6.52, 2.98, 17.16, 0.07, 0.19, -11.7, 1.16, 5.7, 7.11, 29.

In [52]:
from sqlalchemy import select
stmt = select(tb_RAtestbed).where(tb_RAtestbed.그룹 == '*')
print(stmt)

NameError: name 'user_table' is not defined

### 디셈버앤컴퍼니(핀트) 누적 수익률 확인: 37.8%/ 6.1년, 6.7년

In [25]:
import pandas as pd
pd.set_option('display.max_columns', 27)

conn = pymysql.connect(host='localhost', user='root',
    password=DB_SECRET, db='quant_db', charset='utf8')

SQL = "SELECT * FROM quant_db.tb_RAtestbed WHERE 그룹 = '*' and 업체명 = '디셈버앤컴퍼니';"

ra_df_from_sql = pd.read_sql(SQL, conn)

ra_df_from_sql['누적'].mean(), ra_df_from_sql['공시기간'].unique()

/tmp/ipykernel_8966/3023802336.py:9: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  ra_df_from_sql = pd.read_sql(SQL, conn)


(37.878888888888895, array(['6.1년', '6.7년'], dtype=object))

### 쿼터백자산운용 누적 수익률 확인: 32.7% / 6.7년

In [26]:
SQL = "SELECT * FROM quant_db.tb_RAtestbed WHERE 그룹 = '*' and 업체명 = '쿼터백자산운용';"

ra_df_from_sql = pd.read_sql(SQL, conn)
ra_df_from_sql['누적'].mean(), ra_df_from_sql['공시기간'].unique()

/tmp/ipykernel_8966/51686540.py:3: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  ra_df_from_sql = pd.read_sql(SQL, conn)


(32.70166666666666, array(['6.7년'], dtype=object))

### 콴텍 누적 수익률 확인: 15% / 0.7년, 0.9년, 1.4년, 1.7년, 2.4년, 3.7년, 4.4년

In [27]:
SQL = "SELECT * FROM quant_db.tb_RAtestbed WHERE 그룹 = '*' and 업체명 = '콴텍투자일임';"

ra_df_from_sql = pd.read_sql(SQL, conn)
ra_df_from_sql['누적'].mean(), ra_df_from_sql['공시기간'].unique()

/tmp/ipykernel_8966/2944598279.py:3: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  ra_df_from_sql = pd.read_sql(SQL, conn)


(15.057011494252874,
 array(['1.7년', '0.9년', '4.4년', '3.7년', '1.4년', '0.7년', '2.4년'],
       dtype=object))

### 파운트 누적 수익률 확인: 7.8% / 5.2년

In [28]:
SQL = "SELECT * FROM quant_db.tb_RAtestbed WHERE 그룹 = '*' and 업체명 = '파운트투자자문';"

ra_df_from_sql = pd.read_sql(SQL, conn)
ra_df_from_sql['누적'].mean(), ra_df_from_sql['공시기간'].unique()

/tmp/ipykernel_8966/444173559.py:3: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  ra_df_from_sql = pd.read_sql(SQL, conn)


(7.8999999999999995, array(['5.2년'], dtype=object))

### 퀀팃투자자문 누적 수익률 확인: 3.2% / 2.2년

In [29]:
SQL = "SELECT * FROM quant_db.tb_RAtestbed WHERE 그룹 = '*' and 업체명 = '퀀팃투자자문';"

ra_df_from_sql = pd.read_sql(SQL, conn)
ra_df_from_sql['누적'].mean(), ra_df_from_sql['공시기간'].unique()

/tmp/ipykernel_8966/2086324499.py:3: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  ra_df_from_sql = pd.read_sql(SQL, conn)


(3.2233333333333327, array(['2.2년'], dtype=object))

### 크래프트테크놀로지스(&하나은행) 누적 수익률 확인: 2.7% / 6.1년

In [30]:
SQL = "SELECT * FROM quant_db.tb_RAtestbed WHERE 그룹 = '*' and 알고리즘명 like '%크래프트%';"

ra_df_from_sql = pd.read_sql(SQL, conn)
ra_df_from_sql['누적'].mean(), ra_df_from_sql['공시기간'].unique()

/tmp/ipykernel_8966/1067011729.py:3: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  ra_df_from_sql = pd.read_sql(SQL, conn)


(2.723333333333333, array(['6.1년'], dtype=object))

# 코스피 vs. 성장가치 소형주 with 한계기업 vs. 성장가치 소형주 without 한계기업 수익률 비교

In [1]:
import yfinance as yf
import pandas as pd

kospi = yf.download('^KS11', start='2023-01-01', end='2023-07-01', progress=True, auto_adjust=True) # kospi200: ^KS200, kospi: ^KS11
# kospi.head()

(kospi.iloc[-1, 0] * 100 / kospi.iloc[0, 3]) - 100

[*********************100%***********************]  1 of 1 completed


14.95100448162114

In [109]:
kospi.info()

<class 'pandas.core.frame.DataFrame'>
DatetimeIndex: 863 entries, 2020-01-02 to 2023-06-30
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Open    863 non-null    float64
 1   High    863 non-null    float64
 2   Low     863 non-null    float64
 3   Close   863 non-null    float64
 4   Volume  863 non-null    int64  
dtypes: float64(4), int64(1)
memory usage: 40.5 KB


In [110]:
kospi.tail()

Open         High          Low        Close  Volume
Date                                                                  
2023-06-26  2568.649902  2585.310059  2562.030029  2582.199951  479200
2023-06-27  2581.020020  2582.000000  2568.520020  2581.389893  535400
2023-06-28  2590.520020  2590.520020  2559.459961  2564.189941  484200
2023-06-29  2564.020020  2577.689941  2550.020020  2550.020020  602500
2023-06-30  2558.429932  2571.030029  2541.719971  2564.280029  508800

In [76]:
import yfinance as yf
import pandas as pd

kosdaq = yf.download('^KQ11', start='2023-01-01', end='2023-07-01', progress=True, auto_adjust=True) # kosdaq: ^KQ11
# kosdaq.head()

(kosdaq.iloc[-1, 0] * 100 / kosdaq.iloc[0, 3]) - 100

[*********************100%***********************]  1 of 1 completed


28.471653387287063

In [96]:
kosdaq.tail()

Open        High         Low       Close  Volume
Date                                                              
2023-06-26  874.309998  881.239990  869.289978  879.500000    1300
2023-06-27  877.880005  878.429993  867.679993  874.140015    1200
2023-06-28  879.609985  880.229980  866.700012  866.969971    1200
2023-06-29  868.650024  870.750000  859.770020  861.789978    1200
2023-06-30  862.700012  869.239990  853.679993  868.239990    1000

In [73]:
# 한계기업 포함 Top 10

top10_withrisk = ['010770',
 '019540',
 '098660',
 '031510',
 '001210',
 '227100',
 '044480',
 '052300',
 '091440',
 '084870']

In [72]:
# 한계기업 제외 Top 10

top10_withoutrisk = ['021650',
 '031310',
 '046310',
 '032750',
 '006920',
 '001020',
 '290270',
 '012620',
 '203450',
 '051390']

In [113]:
import pandas_datareader.data as web

earning_total_withrisk = 0

for code in top10_withrisk:

    df = web.DataReader(
        code,
        'naver',
        start='20200101',
        end='20230630'
    )

    earning_total_withrisk += (int(df.iloc[-1, 0]) * 100 / int(df.iloc[0, 3])) - 100

earning_total_withrisk = int(earning_total_withrisk) / len(top10_withrisk)

earning_total_withrisk

-35.4

In [114]:
import pandas_datareader.data as web

earning_total_withoutrisk = 0

for code in top10_withoutrisk:

    df = web.DataReader(
        code,
        'naver',
        start='20200101',
        end='20230630'
    )

    earning_total_withoutrisk += (int(df.iloc[-1, 0]) * 100 / int(df.iloc[0, 3])) - 100

earning_total_withoutrisk = int(earning_total_withoutrisk) / len(top10_withoutrisk)

earning_total_withoutrisk

-1.5

# 2022 코스피 수익률 -24.2% / 2023 14.9% / 20 ~ 23.6.30 17.6%

# 2022 성장가치 소형주 & 한계기업 수익률 -38.6% / 2023 6.3% / 20 ~ 23.6.30 -35.4%

# 2022 성장가치 소형주 without 한계기업 수익률 -29.4 / 2023 13.1% / 20 ~ 23.6.30 -1.5%

In [107]:
13.1 * 100 / 6.3

207.93650793650795

In [117]:
(13.1 * 100) / 14.9 - 100

-12.080536912751683